tlatools/src/tlc2/value/impl/IntervalValue.java

```diff
@@ -116,12 +116,13 @@ public final int size() {
 		if (this.high < this.low) {
 			return 0;
 		}
-		final int size = this.high - this.low + 1;
-		if (size < 0) {
-			Assert.fail("Size of interval value exceeds the maximum representable size (32bits)"
-					+ Values.ppr(this.toString()) + ".");
+		try {
+			return Math.addExact(Math.subtractExact(this.high, this.low), 1);
```

Comment:
@Calvin-L Do you have an estimate for how much slower this is going to be?

(PageQueue)

Reply:
Mmm, it's always hard to say anything definitive about small changes like this one. I ran the JMH benchmarks on my laptop for commits 293b755 (this PR) and a6a4c7f (current master). The results are here:

https://gist.github.com/Calvin-L/2d4163c1128a75b7d292158800b01f68

I can't claim to understand how to interpret the results. The average of the "score" fields differs by only about 5% between the two. My intuition says that's just noise, not anything meaningful.

I put together a [microbenchmark](https://gist.github.com/Calvin-L/985f8c442a77b95e076d012140628458) for just the `IntervalValue.size()` method, which suggests that my proposed implementation is about 3-4x slower than the current one. I highly doubt the `size()` method will ever be a bottleneck, but if 3-4x seems steep then we can consider devising some clever-er code to check for overflow.

Reply:
https://jmh.morethan.io/?source=https://gist.githubusercontent.com/Calvin-L/2d4163c1128a75b7d292158800b01f68/raw/617a9f988b1094a58f76312888e40b1a8ee20b94/DiskQueueBenachmark-1579313387-293b755.json is a visualization that is easier to interpret.  I would however suggest to write a dedicated JMH benchmark for IntervalValue#size (i.e. extend tlc2.value.impl.IntervalValueBenchmark), because JMH makes sure to minimize noise created from JVM optimizations.

Reply:
No need to bother (overhead is negligible): https://jmh.morethan.io/?source=https://gist.githubusercontent.com/lemmy/46a8c2ff3bb7c1718ebbcd6c2404230b/raw/0d1a2a92f31618021bb79283534b31380bd24b32/jmh.json

https://gist.github.com/lemmy/46a8c2ff3bb7c1718ebbcd6c2404230b

Will merge PR as is later.